client/app/pages/dashboards/hooks/utils.test.js

```diff
@@ -0,0 +1,131 @@
+import { getAllowedWidgetsForCurrentParam } from "@/pages/dashboards/hooks/utils";
+describe("filterWidgets", () => {
+  test("filter only for allowed widgets per main parameter value", () => {
+    let dashboardParameters = [
+      {
+        name: "controller_param",
+        value: "controller12345",
+      },
+    ];
+    let dashboardAllowedWidgets = {
+      controller12345: ["firstViz", "secondViz"],
+      "4d678ac-cdbc-49c4-ba0f-c8800bce5e8c": ["thirdViz"],
+    };
+    let dashboardAllWidgets = [
+      { id: 1, visualization: { name: "firstViz" } },
+      { id: 2, visualization: { name: "secondViz" } },
+      { id: 3, visualization: { name: "thirdViz" } },
+    ];
+
+    let result = getAllowedWidgetsForCurrentParam(dashboardParameters, dashboardAllowedWidgets, dashboardAllWidgets);
+    expect([
+      { id: 1, visualization: { name: "firstViz" } },
+      { id: 2, visualization: { name: "secondViz" } },
+    ]).toStrictEqual(result);
+  });
+
+  test("filter correctly in case of many parameters", () => {
```

Comment:
Related to my comment about parameters. This isn't clear that the expected behaviour is to match the first parameter